boat-maven-plugin/src/main/java/com/backbase/oss/boat/InputMavenArtifactMojo.java

```diff
@@ -0,0 +1,165 @@
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.impl.ArtifactResolver;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class InputMavenArtifactMojo extends AbstractMojo {
+
+  /**
+   * A maven artifact containing a spec, or multiple to be processed
+   */
+  @Parameter(name = "inputMavenArtifact", property = "inputMavenArtifact", required = false)
+  protected InputArtifact inputMavenArtifact;
+
+  /**
+   * File input used for Linting and Validating
+   * Can be directory or file
+   */
+  @Parameter(name = "input", required = false)
+  protected File input;
+
+  /**
+   * Location of the OpenAPI spec, as URL or local file glob pattern.
+   * <p>
+   * If the input is a local file, the value of this property is considered a glob pattern that must
+   * resolve to a unique file.
+   * </p>
+   * <p>
+   * The glob pattern allows to express the input specification in a version neutral way. For
+   * instance, if the actual file is {@code my-service-api-v3.1.4.yaml} the expression could be
+   * {@code my-service-api-v*.yaml}.
+   * </p>
+   */
+  @Parameter(name = "inputSpec", property = "openapi.generator.maven.plugin.inputSpec", required = false)
+  protected String inputSpec;
+
+  /**
+   * The project being built.
+   */
+  @Parameter(readonly = true, required = true, defaultValue = "${project}")
+  protected MavenProject project;
+
+  /**
+   * Used to set up artifact request
+   */
+  @Parameter(defaultValue = "${repositorySystemSession}", readonly = true)
+  protected RepositorySystemSession repositorySession;
+
+  /**
+   * Used to look up Artifacts in the remote repository.
+   */
+  @Component
+  protected ArtifactResolver artifactResolver;
+
+  /**
+   * List of Remote Repositories used by the resolver.
+   */
+  @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true)
+  protected List<RemoteRepository> remoteRepositories;
+
+
+  @Override
+  public void execute() throws MojoExecutionException, MojoFailureException {
+
+    if (inputMavenArtifact != null && inputMavenArtifact.getArtifactId() !=null){
+      getArtifact();
+    }
+
+    if (input == null && inputSpec== null && inputMavenArtifact == null ){
+      throw new MojoExecutionException("Missing input from plugin, input options are: inputMavenArtifact, input, inputSpec");
+    }
+
+    if (input == null){
+      input = new File(inputSpec);
+    }
+
+  }
+
+
+  private void getArtifact() throws MojoExecutionException {
+    File specUnzipDirectory = new File(project.getBuild().getDirectory()
```

Comment:
Would there be any problem if the same spec was used in multiple plugin executions? We have had some services use the same spec with different executions, for example to generate a client and an API.

Reply:
I just tested it and It seems to be ok, I can't see any issues

Reply:
You have to add a marker to avoid unzipping the archive many times.
Also, take into consideration parallel tasks, the way `getArtifact` is implemented makes the mojos not thread safe.

Reply:
You have to add a marker to avoid unzipping the archive many times.
Also, take into consideration parallel tasks, the way getArtifact is implemented makes the mojos not thread safe.